packages/account/src/providers/transaction-response/transaction-response.ts

```diff
@@ -137,13 +142,20 @@ export class TransactionResponse {
         transactionId: this.id,
       });
 
+      let expectedStatus: GqlTransactionStatusesNames | undefined;
+
       for await (const { statusChange } of subscription) {
         if (statusChange) {
+          expectedStatus = statusChange.type;
           break;
         }
       }
 
-      return this.fetch();
+      return this.fetch(expectedStatus);
+    }
+
+    if (expectedStatusType && response.transaction.status?.type !== expectedStatusType) {
+      return this.fetch(expectedStatusType);
```

Comment:
I agree that there should a `sleep` here, e.g. `sleep(1000)`. I don't think we have to do some polling algorithm. This solution will also soon be superseded by 

The recursive method calls don't look problematic. However, I think the signature of the `fetch` method shouldn't change. This is a temporary solution that will be removed once the underlying problem is resolved. If we go with this implementation, it would be a breaking change to remove the `expectedStatusType` parameter. Making the `expectedStatusType` a private property on the `TransactionResponse` type will enable us to remove it afterwards.

```ts
this.expectedStatus = statusChange.type;
...
if (this.expectedStatus && 
    response.transaction.status?.type !== this.expectedStatus
   ) {  return this.fetch(); }
```

Reply:
@arboleya I did initially have a sleep, however, found the issue had resolved upon the first retry from my testing. Happy to add it back in.

@nedsalk In terms of the breaking change, I will resolve that now.

Reply:
No worries, as @nedsalk said it should be temporary.

Reply:
Upon reconsideration, this can make things too slow. If you have had successful runs without a delay, maybe that is okay temporarily. Or we could try using a smaller delay as well. What do you think?

Reply:
I was pretty happy without the sleep, but maybe a delay of 100ms would be fair.

Reply:
Go for it.

Reply:
761436d6fa863a41052ec5af779a4281b0014f19

Reply:
Sweet. I think we can merge this as is and move on with:
 - https://github.com/FuelLabs/fuels-ts/pull/2901

The above should supersede and undo the changes of this PR, right?

Reply:
Correct - I am happy with that. @nedsalk is that good with you?